### DISCUSSION TITLE
Runtime config gets undefined on build

### DISCUSSION THREAD

**Discussion by vimalmaheedharan** [vercel/next.js#25555] [Help]
### What version of Next.js are you using?

10.2.0

### What version of Node.js are you using?

12.13.1

### What browser are you using?

Chrome

### What operating system are you using?

Windows 10

### How are you deploying your application?

next start

### Describe the Bug

If **ALL** the runtime configuration values are undefined, then build will throw an error if we try to use the value in our application.

In next.config.js
```
module.exports = {  
    publicRuntimeConfig: {  
      staticFolder: undefined,  
      someOther: process.env.SOME_UNDEFINED_VARIABLE  
    },  
  }
```

In pages/index.js

```
import getConfig from 'next/config'

const { publicRuntimeConfig } = getConfig()

export default function Home() {
  return (
    <div className="container">
     ${publicRuntimeConfig.staticFolder}
    </div>
  )
}
```
Build will throw an error here. 

`TypeError: Cannot read property 'staticFolder' of undefined`

But if we have at least one configuration value which is NOT undefined, then this error will not be thrown. See below

```
module.exports = {  
    publicRuntimeConfig: {  
      staticFolder: undefined,  
      someOther: process.env.SOME_UNDEFINED_VARIABLE,  
      yetAnother: 'some value'  
    },  
  }
```

### Expected Behavior

Build should not throw error

### To Reproduce

Clone the repo from - [https://github.com/vimalmahi/next-example](https://github.com/vimalmahi/next-example)

`npm run build`

**Comment by iondrimba** [comment]
@vimalmahi Maybe it has to do with de requirements stated here https://nextjs.org/docs/api-reference/next.config.js/runtime-configuration 

`A page that relies on publicRuntimeConfig must use getInitialProps to opt-out of Automatic Static Optimization. Runtime configuration won't be available to any page (or component in a page) without getInitialProps.`

**Reply by vimal-here** [reply]
@iondrimba - Thanks for looking into this. 

The problem I've stated is a bit different. The error is thrown only when ALL values are null/undefined. If one value is present (any data type), then the error is not thrown. The following will work

```module.exports = {  
    publicRuntimeConfig: {  
      staticFolder: undefined,  
      someOther: process.env.SOME_UNDEFINED_VARIABLE,  
      yetAnother: 'some value'  
    },  
  }
```
If `yetAnother` is not there, then error will be thrown. This issue is **NOT** present in the earlier version I've used - **9.5.0**
